server/personality/personality.module.ts

```diff
@@ -21,20 +26,38 @@ const PersonalityModel = MongooseModule.forFeature([
     },
 ]);
 
-@Module({
-    imports: [
-        PersonalityModel,
-        WikidataModule,
-        ClaimReviewModule,
-        ClaimRevisionModule,
-        HistoryModule,
-        ViewModule,
-        ConfigModule,
-        AbilityModule,
-        CaptchaModule,
-    ],
-    exports: [PersonalityService],
-    providers: [UtilService, PersonalityService, WinstonLogger],
-    controllers: [PersonalityController],
-})
-export class PersonalityModule {}
+@Module({})
+export class PersonalityModule {
+    static register(): DynamicModule {
+        const imports = [];
+        const providers = [personalityServiceProvider];
+        const logger = new Logger("PersonalityModule");
+
+        logger.log(`Database selected: ${dbConfig.type}`);
+
+        if (dbConfig.type === "mongodb") {
+            imports.push(
+                PersonalityModel,
```

Comment:
issue: I believe you don't need to have a variable for imports, no matter what type of DB it's. Also, the only thing you need to push to providers is  `MongoPersonalityService`, there rest should be default.

Reply:
We do need because the PersonalityModel that we are pushing is specific for the MongooseModule use, as you can see on the line 22.